comit/src/ledger/bitcoin.rs

```diff
@@ -0,0 +1,114 @@
+use serde::{de, Deserialize, Deserializer, Serialize};
+use std::fmt;
+
+#[derive(
+    Clone,
+    Copy,
+    Debug,
+    PartialEq,
+    Eq,
+    Hash,
+    Serialize,
+    Deserialize,
+    strum_macros::Display,
+    strum_macros::EnumString,
+)]
+#[serde(rename_all = "lowercase")]
+#[strum(serialize_all = "lowercase")]
+pub enum Bitcoin {
+    Mainnet,
+    Testnet,
+    Regtest,
+}
+
+impl From<Bitcoin> for ::bitcoin::Network {
+    fn from(bitcoin: Bitcoin) -> ::bitcoin::Network {
+        match bitcoin {
+            Bitcoin::Mainnet => ::bitcoin::Network::Bitcoin,
+            Bitcoin::Testnet => ::bitcoin::Network::Testnet,
+            Bitcoin::Regtest => ::bitcoin::Network::Regtest,
+        }
+    }
+}
+
+impl From<::bitcoin::Network> for Bitcoin {
+    fn from(network: ::bitcoin::Network) -> Self {
+        match network {
+            bitcoin::Network::Bitcoin => Bitcoin::Mainnet,
+            bitcoin::Network::Testnet => Bitcoin::Testnet,
+            bitcoin::Network::Regtest => Bitcoin::Regtest,
+        }
+    }
+}
+
+/// Module specifically desgined for use with the `serde(with)` attribute.
+///
+/// # Usage
+///
+/// ```rust
+/// use comit::ledger;
+///
+/// #[derive(serde::Serialize, serde::Deserialize, PartialEq, Debug)]
+/// #[serde(transparent)]
+/// struct Container(#[serde(with = "ledger::bitcoin::bitcoind_jsonrpc_network")] ledger::Bitcoin);
+///
+/// let container = Container(ledger::Bitcoin::Mainnet);
+///
+/// assert_eq!(json_sats, serde_json::to_string(&container).unwrap());
```

Comment:
The variable you are asserting against `json_sats` is never defined.

Also, it should probably not be called "sats" because we are not deserializing an amount here :)